templates/model/spec.go

```diff
@@ -571,6 +577,41 @@ func (s *StringReplacement) UnmarshalYAML(n *yaml.Node) error {
 	return nil
 }
 
+// Append is an action that appends some output to the end of the file.
+type Append struct {
+	// Pos is the YAML file location where this object started.
+	Pos *ConfigPos `yaml:"-"`
+
+	Path String `yaml:"path"`
+	With String `yaml:"with"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler.
+func (s *Append) UnmarshalYAML(n *yaml.Node) error {
+	knownYAMLFields := []string{"path", "with"}
+	if err := extraFields(n, knownYAMLFields); err != nil {
+		return err
+	}
+	type shadowType Append
+	shadow := &shadowType{} // see "Q2" in file comment above
+
+	if err := n.Decode(shadow); err != nil {
+		return err
+	}
+	*s = Append(*shadow)
+	s.Pos = yamlPos(n)
+
+	return nil
+}
+
+// Validate implements Validator.
+func (s *Append) Validate() error {
+	return errors.Join(
+		notZero(s.Pos, s.Path, "path"),
```

Comment:
Do these validations make sense? I'm assuming we want it to be an error if missing rather than noop.

Reply:
Yeah, this looks good to me.